insights/parsers/nova_user_ids.py

```diff
@@ -0,0 +1,75 @@
+"""
+Get ``uid`` of user ``nova`` and ``nova_migration``
+===================================================
+
+The parser class in this module uses base parser class
+``CommandParser`` to get uids of the user ``nova`` and
+``nova_migration``.
+
+Parsers included in this module are:
+
+NovaUID - command ``id -u nova``
+--------------------------------
+
+NovaMigrationUID - command ``id -u nova_migration``
+---------------------------------------------------
+"""
+from insights import CommandParser, parser
+from insights.parsers import ParseException, SkipException
+from insights.specs import Specs
+
+
+@parser(Specs.nova_uid)
+class NovaUID(CommandParser):
+    '''Parse output of ``id -u nova`` and get the ``uid`` (int).
+
+    Typical output of the ``id -u nova`` command is::
+
+        162
+
+    However the id number may vary.
+
+    Examples:
+
+        >>> nova_uid.data
+        162
+
+    Attributes:
+        data: ``int`` if 'nova' user exist.
+
+    Raises:
+        SkipException: If 'nova' user not found or output is empty.
+        ParseException: If multi-line output. Output of such kind is not yet expected from the command `id`.
+    '''
+    def parse_content(self, content):
+        if not content or (len(content) == 1 and "no such user" in content[0]):
+            raise SkipException()
+        elif len(content) == 1 and ("no such user" not in content[0]) and content[0].isdigit():
+            self.data = int(content[0])
+        elif len(content) > 1:
+            raise ParseException("Unable to parse user ID")
```

Comment:
@psachin I believe this should just be an `else:`.  If len(content) == 1 and content[0] is not a number that should still be a ParseException shouldn't it?  You shouldn't get a uid value that is not a number should you?

Reply:
@bfahr  Yes. For the last one the o/p should not be a number(hopefully). I agree with your suggestion and better raise an exception in `else`. Thanks